### PR TITLE
feat: combine adaptive expression types

### LINKED ISSUE #3193: Combine Adaptive Expression types
See [this comment](https://github.com/microsoft/botbuilder-js/pull/3182#discussion_r557802555), but it would be good to combine:

```typescript
export interface BaseTeamsCacheInfoResponseDialogConfiguration {
    disabled?: boolean | string | BoolExpression;
```

using:

```typescript
type BoolExpressionUnion = boolean | string | BoolExpression;
```

so it becomes:

```typescript
export interface BaseTeamsCacheInfoResponseDialogConfiguration {
    disabled?: BoolExpressionUnion;
```

This occurs in a LOT of the JS Adaptive- libraries.

### PR DESCRIPTION
**Pull request by joshgummersall**
Fixes #3193

### CODE REVIEW

File: libraries/botbuilder-dialogs-adaptive-teams/src/actions/baseAuthResponseDialog.ts [hunk 1]
```diff
@@ -44,10 +45,8 @@ export abstract class BaseAuthResponseDialog
      */
     public title?: StringExpression;
 
-    public getConverter(property: keyof BaseAuthResponseDialogConfiguration | string): Converter | ConverterFactory {
+    public getConverter(property: keyof BaseAuthResponseDialogConfiguration): Converter | ConverterFactory {
         switch (property) {
-            case 'disabled':
-                return new BoolExpressionConverter();
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs-adaptive-teams/src/actions/baseAuthResponseDialog.ts:50]: Caught by `super`

File: libraries/botbuilder-dialogs-adaptive-teams/src/actions/baseSendTaskModuleContinueResponse.ts [hunk 2]
```diff
@@ -52,12 +53,8 @@ export abstract class BaseSendTaskModuleContinueResponse
      */
     public completionBotId?: StringExpression;
 
-    public getConverter(
-        property: keyof BaseSendTaskModuleContinueResponseConfiguration | string
-    ): Converter | ConverterFactory {
+    public getConverter(property: keyof BaseSendTaskModuleContinueResponseConfiguration): Converter | ConverterFactory {
         switch (property) {
-            case 'disabled':
-                return new BoolExpressionConverter();
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs-adaptive-teams/src/actions/baseSendTaskModuleContinueResponse.ts:60]: Caught by `super`

File: libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendAppBasedLinkQueryResponse.ts [hunk 3]
```diff
@@ -50,8 +48,6 @@ export class SendAppBasedLinkQueryResponse
 
     public getConverter(property: keyof SendAppBasedLinkQueryResponseConfiguration): Converter | ConverterFactory {
         switch (property) {
-            case 'disabled':
-                return new BoolExpressionConverter();
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendAppBasedLinkQueryResponse.ts:54]: Caught by `super`

File: libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendAppBasedLinkQueryResponse.ts [hunk 4]
```diff
@@ -6,30 +6,28 @@
...
     TemplateInterface,
 } from 'botbuilder-dialogs';
-import { ActivityTemplateConverter } from 'botbuilder-dialogs-adaptive/lib/converters';
-import { BaseTeamsCacheInfoResponseDialog } from './baseTeamsCacheInfoResponseDialog';
 
-export interface SendAppBasedLinkQueryResponseConfiguration extends DialogConfiguration {
-    disabled?: boolean | string | BoolExpression;
-    property?: string | Expression | StringExpression;
-    card?: TemplateInterface<Activity, DialogStateManager>;
+export interface SendAppBasedLinkQueryResponseConfiguration extends BaseTeamsCacheInfoResponseDialogConfiguration {
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendAppBasedLinkQueryResponse.ts:28]: Configuration should extend if we are inheriting from another dialog.

File: libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendMEAuthResponse.ts [hunk 5]
```diff
@@ -7,13 +7,13 @@
...
 
 import { Activity, CardAction, MessagingExtensionResult } from 'botbuilder';
+import { BaseAuthResponseDialog } from './baseAuthResponseDialog';
 import { DialogContext } from 'botbuilder-dialogs';
-import { BaseAuthResponseDialog, BaseAuthResponseDialogConfiguration } from './baseAuthResponseDialog';
 
 /**
  * Send a messaging extension 'result' response when a Teams Invoke Activity is received with activity.name='composeExtension/queryLink'.
  */
-export class SendMEAuthResponse extends BaseAuthResponseDialog implements BaseAuthResponseDialogConfiguration {
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendMEAuthResponse.ts:16]: No extra configuration, so `implements T` is implied by inheriting from base classs.

File: libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendTabCardResponse.ts [hunk 6]
```diff
@@ -37,14 +38,8 @@ export class SendTabCardResponse extends BaseAuthResponseDialog implements BaseA
      */
     public cards?: TemplateInterface<Activity, DialogStateManager>;
 
-    public getConverter(property: keyof BaseAuthResponseDialogConfiguration | string): Converter | ConverterFactory {
+    public getConverter(property: keyof SendTabCardResponseConfiguration): Converter | ConverterFactory {
         switch (property) {
-            case 'disabled':
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendTabCardResponse.ts:42]: All handled by `super`

**Review by stevengum — APPROVED** [review]

**Pull request merged**